src/flint/fmpq.jl

```diff
@@ -387,18 +387,9 @@ isless(a::QQFieldElem, b::Float64) = isless(BigFloat(a), b)
 #
 ###############################################################################
 
-function ^(a::QQFieldElem, b::Int)
+function ^(a::QQFieldElemOrPtr, b::IntegerUnionOrPtr)
   iszero(a) && b < 0 && throw(DivideError())
```

Comment:
```suggestion
  iszero(a) && is_negative(b) && throw(DivideError())
```
since `b` can be a Ptr, `b < 0` is actually different from `is_negative(b)`. This is also relevant for all other places where the exponent may now be a Ptr

Reply:
Let's keep this open since there are a few other instances of this issue that are not fixed yet. Just for both of us to not forget